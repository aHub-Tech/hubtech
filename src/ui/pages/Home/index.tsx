import Main from 'ui/components/Main';
import * as S from './styles';

export default function Home() {
  return (
    <S.Wrapper>
      <Main />
    </S.Wrapper>
  );
}
