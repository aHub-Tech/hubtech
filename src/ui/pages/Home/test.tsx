import { render, screen } from '@testing-library/react';

import Home from '.';

jest.mock('ui/components/Main', () => {
  return {
    __esModule: true,
    default: function MainMock() {
      return <div data-testid="main-component"></div>;
    }
  };
});

describe('<Home />', () => {
  it('should render the <Main /> component', () => {
    const { container } = render(<Home />);

    expect(screen.getByTestId('main-component')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
