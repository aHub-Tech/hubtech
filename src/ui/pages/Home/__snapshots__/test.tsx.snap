// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the <Main /> component 1`] = `
.c0 {
  margin: 20px;
}

<div
  class="c0"
>
  <div
    data-testid="main-component"
  />
</div>
`;
